helm-v3/007-security.md

```diff
@@ -13,4 +13,116 @@ Helm Classic). The following security goals are met by this proposal:
+existing key (if present) or by the newly generated key.
+
+### Adding Public Keys to Repositories
+
+When a chart is added to the index using `helm index`, the public key will be
+included in a designated section of the index.
+
+```yaml
+apiVersion: v2
+entries:
+  brigade:
+  - apiVersion: v1
+    appVersion: v0.12.0
+    created: 2018-03-20T21:20:29.430663118Z
+    description: Brigade provides event-driven scripting of Kubernetes pipelines.
+    digest: 90e1cdef4303e39fb105c3a0f745a2d802bb44deaffa29f9c25933fcd53ab262
+    name: brigade
+    urls:
+    - brigade-0.12.0.tgz
+    version: 0.12.0
+  - apiVersion: v1
+    appVersion: v0.11.0
+    created: 2018-03-20T21:20:29.430245427Z
+    description: Brigade provides event-driven scripting of Kubernetes pipelines.
+    digest: 40bc429edc50b66cd8ab51f419d0652b51d2aed5804fc2769b781b833168c582
+    name: brigade
+    urls:
+    - brigade-0.11.0.tgz
+    version: 0.11.0
+keys:
+  - |-
+    -----BEGIN PGP PUBLIC KEY BLOCK-----
+    Comment: GPGTools - https://gpgtools.org
+
+    mQINBFd1LD8BEADJTqMXYG74qqmgZFjakinAJIzViJ0KmXkI+mu4S6SCwz/QuBVW
+    CqnNshIfrdP3rB+LiDFybizjO9bHVTV46tZhbHGI6QWsEPnVoorZEO4v9lXGESrx
+    M1EIwzvzRD2VdWaucMqLpR6YAShmiOXa2qng1OkWXFTqeq+2p8MtX19DKzhBS7r6
+    +fcTODX6yyVUt1b1DO0JU+hwGJ8yNEWESp7gu7XE6rxKI+HAPhiL8oyXUIORlzAb
+    p00w9I5v5KMKZcCmr09wTwvxq1DxyBYK3GGvlfWvrbK3n6iXWDYGVQf4Tcjx7A/3
+    FAEriJfs8sP6lMq22b7ZsDVqJafOMeaCQTB8cmuEVHt/OKcOexpQ58AWIKyXCehf
+    # ...
+    PJNSLa0HO7J8vTnZOrpBFOBrneu4pjg8Z+Ozf7QgmH55BHRIzmqtiY2ZlsfxsCmI
+    /4IJTxm9LpsVvzr4ZilExyFISkawKOt5P1P8B4VyORrHvn/tF8YcEliNmlXGeW+P
+    KFdxAk9vWD0+F3pxXt5OVxbpAWGErKWugN6o04+yXrAI26OMI0FUAi8sWOsFbhGi
+    awRB1rHuMB+3gnqAFN4lY3f9pqxh9WVXZVf08JAPheD+bb+aSlEX22sETIhylu3W
+    oV8SbEZ/NmSbnbI1LNHPw7UugdVFbfxyZWX/+ll/KdFmFQclPrbQrNFs26vqtCCy
+    ed3bZYsGOMVGhMojg2zdcCmJSLQN
+    =bw6S
+    -----END PGP PUBLIC KEY BLOCK-----
+  - |-
+    -----BEGIN PGP PUBLIC KEY BLOCK-----
+    Comment: GPGTools - https://gpgtools.org
+
+    mQINBFd1LD8BEADJTqMXYG74qqmgZFjakinAJIzViJ0KmXkI+mu4S6SCwz/QuBVW
+    CqnNshIfrdP3rB+LiDFybizjO9bHVTV46tZhbHGI6QWsEPnVoorZEO4v9lXGESrx
+    M1EIwzvzRD2VdWaucMqLpR6YAShmiOXa2qng1OkWXFTqeq+2p8MtX19DKzhBS7r6
+    +fcTODX6yyVUt1b1DO0JU+hwGJ8yNEWESp7gu7XE6rxKI+HAPhiL8oyXUIORlzAb
+    p00w9I5v5KMKZcCmr09wTwvxq1DxyBYK3GGvlfWvrbK3n6iXWDYGVQf4Tcjx7A/3
+    FAEriJfs8sP6lMq22b7ZsDVqJafOMeaCQTB8cmuEVHt/OKcOexpQ58AWIKyXCehf
+    # ...
+    PJNSLa0HO7J8vTnZOrpBFOBrneu4pjg8Z+Ozf7QgmH55BHRIzmqtiY2ZlsfxsCmI
+    /4IJTxm9LpsVvzr4ZilExyFISkawKOt5P1P8B4VyORrHvn/tF8YcEliNmlXGeW+P
+    KFdxAk9vWD0+F3pxXt5OVxbpAWGErKWugN6o04+yXrAI26OMI0FUAi8sWOsFbhGi
+    awRB1rHuMB+3gnqAFN4lY3f9pqxh9WVXZVf08JAPheD+bb+aSlEX22sETIhylu3W
+    oV8SbEZ/NmSbnbI1LNHPw7UugdVFbfxyZWX/+ll/KdFmFQclPrbQrNFs26vqtCCy
+    ed3bZYsGOMVGhMojg2zdcCmJSLQN
+    =bw6S
+    -----END PGP PUBLIC KEY BLOCK-----
+```
+
+During `helm repo add` and `helm repo update`, the user will be prompted to accept
```

Comment:
How will this work for the default `stable` repo? Perhaps the prompt to accept the key could happen at `helm init` time?

Reply:
Option A: User is prompted at `helm init`.

Option B: A stubbed repo with the public keys is delivered in the Helm binary.

Reply:
I don't have a strong opinion on this either way, Option B is less work for the user and similar to what we do today, but Option A is much more transparent and gives you the option to not add the stable repo (which may be something people want if they only want to point to private repos/are behind a firewall).